### PR TITLE
quick fix for GKE cluster creation in case of its status is ERROR

### LINKED ISSUE #2055: GKE cluster creation failed due to its status is ERROR
**Describe the bug**
If a GKE cluster status is error after creation, the posthook will start and fail.

**Steps to reproduce the issue:**
Sometimes it happens.

**Expected behavior**
If cluster status was ERROR, cluster creation should return error.

**Additional context**
```
status: ERROR
statusMessage: Failed to create cluster
subnetwork: default
```

### PR DESCRIPTION
**Pull request by pbalogh-sa**
| Q               | A
| --------------- | ---
| Bug fix?        |yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #2055
| License         | Apache 2.0

### What's in this PR?
Return error if the cluster status after GKE cluster creation is ERROR 
### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)

### CODE REVIEW

File: cluster/gke.go [hunk 1]
```diff
@@ -321,6 +322,10 @@ func (c *GKECluster) CreateCluster() error {
 		// TODO status code !?
 		return errors.New(be.Message)
 	}
+	if gkeCluster.Status == statusError {
+		be := getBanzaiErrorFromError(err)
+		return errors.New(be.Message)
```
Comment (stoader) [inline cluster/gke.go]: I'd include also the status message provided by `gkeCluster` or if we don't want to display that to the user than log the status message.

File: cluster/gke.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ import (
 const (
 	statusRunning = "RUNNING"
 	statusDone    = "DONE"
+	statusError   = "ERROR"
```
Comment (orymate) [inline cluster/gke.go]: these are operation statuses. the cluster status can be used from `pkg/cluster.Error`

File: cluster/gke.go [hunk 3]
```diff
@@ -321,6 +322,10 @@ func (c *GKECluster) CreateCluster() error {
 		// TODO status code !?
 		return errors.New(be.Message)
 	}
+	if gkeCluster.Status == statusError {
```
Comment (orymate) [inline cluster/gke.go]: in which cases will `err == nil` be true (i.e. the function not returning in line 323), but the error extraction below work?

File: cluster/gke.go (hunk 1 above)
Comment (orymate) [reply cluster/gke.go]: this line will always be equivalent to `return errors.New("")`, won't it?

Comment (pbalogh-sa) [reply cluster/gke.go]: @orymate , you are right I will fix it

File: cluster/gke.go (hunk 3 above)
Comment (pbalogh-sa) [reply cluster/gke.go]: If getClusterGoogle func return without any error and the gkeCluster.Status is ERROR.
`gkeCluster, err := getClusterGoogle(svc, cc)`

**Pull request merged**